endpoint/loadbalancer/loadbalancer.go

```diff
@@ -0,0 +1,163 @@
+package loadbalancer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/backube/pvc-transfer/endpoint"
+	"github.com/backube/pvc-transfer/utils"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type loadBalancer struct {
+	logger logr.Logger
+
+	hostname        string
+	ingressPort     int32
+	backendPort     int32
+	namespacedName  types.NamespacedName
+	labels          map[string]string
+	ownerReferences []metav1.OwnerReference
+}
+
+// Register should be used as soon as scheme is created to add
+// route objects for encoding/decoding
+func Register(scheme *runtime.Scheme) error {
```

Comment:
What do you think of renaming `Register()` to `AddToScheme()`?
That's its function and the signature matches... The only thing strange if that it's not adding a specific group, its adding all the groups that the implementation depends on.